htdocs/js/ui_utils.js

```diff
@@ -182,6 +183,41 @@ ui_utils.install_common_ace_key_bindings = function(widget, get_language) {
                         shell.scroll_to_end();
                     });
             }
+        },
+
+        {
+            name: 'execute-cursor-at-beginning',
```

Comment:
so `execute-` isn't some kind of boilerplate prefix, it is that those other commands actually execute R code. the names could just be "beginning of line" "end of line"